tests/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+# OpenAssetIO Dependency.
+find_package(OpenAssetIO REQUIRED)
+enable_testing()
+
+add_executable(test.cpp test.cpp)
+add_test(cpp test.cpp)
```

Comment:
[consider] (Apologies for saying otherwise offline, but) I think for this test in particular, it would be worth running against the install tree rather than build tree, since the test is largely about checking the expected directory structure, which may be different in the install tree if we've messed up somewhere.

Reply:
After being reminded in offline discussion, the test for the install tree may well be covered by augmenting the OpenAssetIO-Test-CMake project to pull in this repo as an external dependency. So I'll downgrade this to [consider][minor].

Reply:
Def +1 on testing the install tree.

Reply:
I implemented this, on advice. However, it became apparent after that this really isn't testing anything, due to the header-only nature of the project, unlike projects that have dynamic libs. All we can do is build the test from the headers in the build-dir, copy the executable to the install dir, and then run it from there.

David and I spoke, and the best alternative to my mind is to have a fully isolated test project in this repo, however, as the test-cmake project is being [altered](https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/pull/7) to cover that sort of integration test, I don't think there's any need to change things here.

I've dropped the commits that add the install-tree test changes (although I won't push them yet to avoid breaking code-review links)